Makefile

```diff
@@ -245,27 +249,48 @@ CONTROLLER_TOOLS_VERSION ?= v0.11.1
 
 KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
 .PHONY: kustomize
-kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary.
+kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary. If wrong version is installed, it will be removed before downloading.
```

Comment:
I tested this locally, and the version check only executes if the $(KUSTOMIZE) target is phony. You can add this as a separate line, or append "$(KUSTOMIZE)" to L251.

Reply:
Argh, I hate github's PR interface.

But you get the idea: you need to make the KUSTOMIZE target phony.

Reply:
ok, noted but this is pulled directly from the skeleton Makefile generated by operator-sdk and I would prefer not to include other improvements in these standardize_makefile pull requests unless it is an issue with how it is currently?

Reply:
It is an issue, because the version logic doesn't get triggered unless KUSTOMIZE is phony:

```
# Start clean
[cinder-operator]$ rm bin/kustomize

# Install an old version (v3.8.6)
[cinder-operator]$ KUSTOMIZE_VERSION=v3.8.6 make kustomize
test -s /home/abishop/cinder-operator/bin/kustomize || { curl -Ss "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash -s -- 3.8.6 /home/abishop/cinder-operator/bin; }
{Version:kustomize/v3.8.6 GitCommit:c1747439cd8bc956028ad483cdb30d9273c18b24 BuildDate:2020-10-29T23:07:50Z GoOs:linux GoArch:amd64}
kustomize installed to /home/abishop/cinder-operator/bin/kustomize

[cinder-operator]$ bin/kustomize version
{Version:kustomize/v3.8.6 GitCommit:c1747439cd8bc956028ad483cdb30d9273c18b24 BuildDate:2020-10-29T23:07:50Z GoOs:linux GoArch:amd64}

# This should trigger the upgrade to KUSTOMIZE_VERSION (v3.8.7)
[cinder-operator]$ make kustomize
make: Nothing to be done for 'kustomize'.

# Repeat with phony KUSTOMIZE target. Now the upgrade works as intended.
[cinder-operator]$ git diff
diff --git a/Makefile b/Makefile
index 0e50536..02f39d8 100644
--- a/Makefile
+++ b/Makefile
@@ -248,7 +248,7 @@ KUSTOMIZE_VERSION ?= v3.8.7
 CONTROLLER_TOOLS_VERSION ?= v0.11.1
 
 KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
-.PHONY: kustomize
+.PHONY: kustomize $(KUSTOMIZE)
 kustomize: $(KUSTOMIZE) ## Download kustomize locally if necessary. If wrong version is installed, it will be removed before downloading.
 $(KUSTOMIZE): $(LOCALBIN)
        @if test -x $(LOCALBIN)/kustomize && ! $(LOCALBIN)/kustomize version | grep -q $(KUSTOMIZE_VERSION); then \

[cinder-operator]$ make kustomize
/home/abishop/cinder-operator/bin/kustomize version is not expected v3.8.7. Removing it before installing.
test -s /home/abishop/cinder-operator/bin/kustomize || { curl -Ss "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash -s -- 3.8.7 /home/abishop/cinder-operator/bin; }
{Version:kustomize/v3.8.7 GitCommit:ad092cc7a91c07fdf63a2e4b7f13fa588a39af4f BuildDate:2020-11-11T23:14:14Z GoOs:linux GoArch:amd64}
kustomize installed to /home/abishop/cinder-operator/bin/kustomize

[cinder-operator]$ bin/kustomize version
{Version:kustomize/v3.8.7 GitCommit:ad092cc7a91c07fdf63a2e4b7f13fa588a39af4f BuildDate:2020-11-11T23:14:14Z GoOs:linux GoArch:amd64}
```